src/Tools/Source/CompilerGeneratorTools/Source/BoundTreeGenerator/BoundNodeClassWriter.cs

```diff
@@ -842,13 +842,14 @@ private void WriteField(Field field)
                     else if (field.Override)
                     {
                         // We emit a suppression here because the bound nodes use a pattern which is safe, but can't be tracked.
+                        var suppression = FieldNullHandling(node, field.Name) is (NullHandling.Allow or NullHandling.Always) ? "" : "!";
 
                         // The point of overriding a property is to change its nullability, usually
                         // from nullable to non-nullable. The base is annotated as nullable,
                         // but since the base property is always set via the base call in the
                         // constructor, as long as the parameter to the current class's constructor is not
                         // nullable, the base property is always non-null.
-                        WriteLine($"public new {field.Type} {field.Name} => base.{field.Name}!;");
+                        WriteLine($"public new {field.Type} {field.Name} => base.{field.Name}{suppression};");
```

Comment:
>new [](start = 43, length = 3)

This looks surprizing to me. This isn't an override.

Reply:
> This isn't an override.

`BoundExpression.Type` is not `virtual`. It could be. But that was not the intent of this change.

---
In reply to: [603320609](https://github.com/dotnet/roslyn/pull/52203#discussion_r603320609) [](ancestors = 603320609)